src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java

```diff
@@ -730,7 +730,30 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 						jarArchiver.addDirectory(file);
 					} else {
 						getLog().info("merge artifact: " + entry.getKey());
-						jarArchiver.addArchivedFileSet(file);
+						
+						// Respect filter if set
+						String filter = entry.getValue().filter;
+						if(filter == null) {
+							jarArchiver.addArchivedFileSet(file);
+						} else {
+						    
+						    // Filter elements must be separated int two lists
+						    List<String> includes = new ArrayList<String>();
+						    List<String> excludes = new ArrayList<String>();
+
+						    // Elements starting with ! should be excluded while others should be included
+						    for(String element : filter.split(",")) {
+							if(element.startsWith("!")) {
+							    excludes.add(element.substring(1));
+							}else {
+							    includes.add(element);
+							}
+						    }
+
+						    jarArchiver.addArchivedFileSet(file,
```

Comment:
It is not deprecated yet and also not in newest maven-archiver version 3.5.0 which includes plexus-archiver 2.4.
I can change my code to use the new method if you want.

Btw. maven-archiver used in this plugin is more than ten years old:
https://search.maven.org/artifact/org.apache.maven/maven-archiver

Reply:
`addArchivedFileSet(File archiveFile,
                                 String[] includes,
                                 String[] excludes)`
is deprecated in that link, but
`addArchivedFileSet(ArchivedFileSet fileSet)`
is not.

Not sure since which version that is though, probably the newest. (4.2.2?)
Anyway, not necessary to change that since the new method might not even exist in the version we use.

Reply:
You are right, the link points to 4.2.2 where the method mentioned is deprecated.
Regarding to my IDE the used version in proguard-maven-plugin is much older.
![image](https://user-images.githubusercontent.com/19415000/91041922-2eaacf00-e611-11ea-8892-df7d8c86077e.png)

I did a quick test by changing the dependency of maven-archiver to 3.5.0 but this leads to compile errors and more deprecated code.